### ISSUE TITLE
PDB: Empty data structures

### ISSUE THREAD

**Issue by kterhorst** [NationalSecurityAgency/ghidra#183]
Hello, Ghidra is a great tool and it uses the information provided in the pdb/xml but in some cases it doesn't recognize data structures. For example, when I hover over some type it states "WARNING! Empty Structure". I checked the xml and all the type information is there. It works for some types and doesn't for others. I checked the source code and can't quite figure out how to force Ghidra to use all type information. I can't post an example file (due to legal restrictions) but I am sure some of you guys have seen this problem before. Is there a way to fix this behaviour? The exe file is 4.8 MB and the PDB is 96 MB and I am running Ghidra on a machine with 16 GB of ram - I don't think lack of memory is the problem.

**Comment by ghidra1** [comment]
Check for multiple definitions of the same structure by using the Datatype Manager tree filter.   Those created by the PDB have a common root category.  If the PDB's version is correct within the tree and another empty one exists, drag the good one onto the empty one to replace all uses of the empty one.

We have yet to address reconciling multiple definitions of the same data type.

**Comment by kterhorst** [comment]
Hi. Unfortunately this didn't do the trick. I tried to delete redundant definitions of the same data structures manually in the XML file - didn't work either. I chose one structure (which wasn't recognized) with only two fields, both of them being C++ template instances, and changed their types to plain "int" and voilà - the members were detected properly. I will wait until the source code of Ghidra is released and then I am going to give this problem a closer look.

**Comment by ghidra1** [comment]
You may have misunderstood my previous post.  Structure placeholder definitions may also be introduced into the datatype manager during demangling.  At present, the demangler does not attempt to utilize the types defined by the PDB import which are stored under a pdb datatype category folder.  Using the datatype tree GUI, alternate definitions within the tree can be replaced with those that may have been filled-out user the pdb category.  Using the tree filter, type the name of a type to see if multiple definitions exist then drag the good onto the bad definition to perform a replacement.

**Comment by kterhorst** [comment]
I don't think I misunderstood your post. I chose a class whose structure is empty. Then I used the filter in the data type manager to see if there are multiple definitions. But there is only _one_ definition of the structure in the data type manager. However, in the xml file there are multiple definitions of the structure and both are filled with its correct members. I invited you to a private repository with a test case.
